packages/ckeditor5-clipboard/tests/dragdropexperimental.js

```diff
@@ -0,0 +1,2241 @@
+				} );
+
+				expectDraggingMarker( model.createPositionAt( modelWidget, 'after' ) );
+
+				env.isGecko = originalEnvGecko;
+			} );
+		} );
+
+		describe( 'dragend', () => {
+			it( 'should reset block dragging when dropped outside the editable', () => {
+				setModelData( model,
+					'<paragraph>foobar</paragraph>' +
+					'[<horizontalLine></horizontalLine>]'
+				);
+
+				const plugin = editor.plugins.get( 'DragDropExperimental' );
+				const modelElement = model.document.getRoot().getNodeByPath( [ 1 ] );
+				const viewElement = mapper.toViewElement( modelElement );
+
+				// Fire the 'dragstart' event to change the '_blockMode to true
+				viewDocument.fire( 'dragstart', {
+					domTarget: domConverter.mapViewToDom( viewElement ),
+					target: viewElement,
+					domEvent: {},
+					dataTransfer: createDataTransfer( {} ),
+					preventDefault: sinon.spy(),
+					stopPropagation: sinon.spy()
+				} );
+
+				// Fire the 'dragend' event on the document
+				document.dispatchEvent( new Event( 'dragend' ) );
+
+				// Check if the blockMode changes to 'false'
+				expect( plugin._blockMode ).to.be.false;
+				expect( model.markers.has( 'drop-target' ) ).to.be.false;
+			} );
+		} );
+
+		describe( 'drop', () => {
+			// TODO: to be discussed.
+			it.skip( 'should update targetRanges', () => {
+				setModelData( model,
+					'<paragraph>[]foobar</paragraph>' +
+					'<horizontalLine></horizontalLine>'
+				);
+
+				const dataTransferMock = createDataTransfer();
+				const spyClipboardInput = sinon.spy();
+
+				viewDocument.on( 'clipboardInput', ( event, data ) => spyClipboardInput( data ) );
+
+				dataTransferMock.dropEffect = 'move';
+				const targetPosition = model.createPositionAt( root.getChild( 1 ), 0 );
+				fireDrop( dataTransferMock, targetPosition );
+
+				expect( spyClipboardInput.called ).to.be.true;
+
+				const data = spyClipboardInput.firstCall.firstArg;
+				expect( data.method ).to.equal( 'drop' );
+				expect( data.dataTransfer ).to.equal( dataTransferMock );
+				expect( data.targetRanges.length ).to.equal( 1 );
+				expect( data.targetRanges[ 0 ].isEqual( view.createRangeOn( viewDocument.getRoot().getChild( 1 ) ) ) ).to.be.true;
+			} );
+		} );
+	} );
+
+	describe( 'integration with the WidgetToolbarRepository plugin', () => {
+		let editor, widgetToolbarRepository, editorElement, viewDocument;
+
+		beforeEach( () => {
+			editorElement = document.createElement( 'div' );
+			document.body.appendChild( editorElement );
+
+			return ClassicTestEditor
+				.create( editorElement, {
+					plugins: [ Paragraph, WidgetToolbarRepository, DragDropExperimental, HorizontalLine ]
+				} )
+				.then( newEditor => {
+					editor = newEditor;
+					viewDocument = editor.editing.view.document;
+					widgetToolbarRepository = editor.plugins.get( WidgetToolbarRepository );
+
+					editor.setData( '<p></p>' );
+				} );
+		} );
+
+		afterEach( () => {
+			editorElement.remove();
```

Comment:
Before `editor.destroy()`? In other `afterEach()` the order is reversed. If both are correct, pick the consistent style. If not - fix this or other ones.

Reply:
It also was copied from `dragdrop.js`